lib/Listener/GroupDeletedListener.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace OCA\Photos\Listener;
+
+use OCA\Photos\Album\AlbumMapper;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\Group\Events\GroupDeletedEvent;
+
+class GroupDeletedListener implements IEventListener {
+	private AlbumMapper $albumMapper;
+
+	public function __construct(AlbumMapper $albumMapper) {
+		$this->albumMapper = $albumMapper;
+	}
+
+	public function handle(Event $event): void {
+		if (!($event instanceof GroupDeletedEvent)) {
+			return;
+		}
+        
+        // Get all shared albums for this group:
+        $albums_group = $this->albumMapper->getSharedAlbumsForCollaboratorWithFiles($event->getGroup()->getGID(), AlbumMapper::TYPE_GROUP);
+
+        // Get all users of this group:
+        $users = $event->getGroup()->getUsers();
+
+        foreach ($users as $user) {
```

Comment:
The algo seem convoluted.

- You have the list of albums for a group
- You have the list of users for a group
- For each albums:
    - Remove the group from the collaborators list
    - For each user
         - If the use is not in the collaborator list, remove its files.

I don't think you need to query the DB more than once.

Let me know if I am missing something.

Reply:
> * If the use is not in the collaborator list, remove its files.

This is the culprit - in order to know if the user is collaborator of a given album, I need to fetch this - so one more DB query for every album. Whether I loop over users or albums doesn't matter.

Am I missing something obvious? :thinking:

Reply:
Ah, I thought that the list of collaborators could be retrieved from `AlbumInfo`. But it is not. Sorry for the trouble !

Reply:
In this case, adding a new method `getAlbumsForCollaboratorId` to `AlbumMapper` is necessary to at least reduce the DB load.

Reply:
I agree, and that's done as suggested by you above. Not sure why it doesn't show here.